src/api/ingestion.ts

```diff
@@ -21,16 +27,32 @@ function parsePdfIngestionResponse(response: IngestResponse): ReferenceItem[] {
   }));
 }
 
-export async function runPDFIngestion(): Promise<ReferenceItem[]> {
-  const uploadsDir = await getSystemPath(getUploadsDir());
-  const response = await callSidecar('ingest', { pdf_directory: String(uploadsDir) });
-  return parsePdfIngestionResponse(response);
+export async function runPDFIngestion(projectId?: string): Promise<ReferenceItem[]> {
+  if (import.meta.env.VITE_IS_WEB) {
```

Comment:
I don't really like having to make this check in each function. I would rather define an `IngestionAPI` abstract interface, with the methods we need, and then create 2 implementations `WebIngestionAPI` and `SidecarIngestionAPI` of this interface and export the correct implementation depending on `VITE_IS_WEB`

Reply:
I haven't implement that bc we will remove the calls via sidecar for the references/ingestion process. Maybe I should make them into this branch actually!